plugin/helper/writer.go

```diff
@@ -73,16 +74,39 @@ func (w *WriterPlugin) SetOutputs(plugins []plugin.Plugin) error {
 	for _, pluginID := range w.OutputIDs {
 		plugin, ok := w.findPlugin(plugins, pluginID)
 		if !ok {
-			return fmt.Errorf("plugin `%s` does not exist", pluginID)
+			return fmt.Errorf("plugin '%s' does not exist", pluginID)
 		}
 
 		if !plugin.CanProcess() {
-			return fmt.Errorf("plugin `%s` can not process entries", pluginID)
+			return fmt.Errorf("plugin '%s' can not process entries", pluginID)
 		}
 
 		outputPlugins = append(outputPlugins, plugin)
 	}
 
+	// No outputs have been set, so use the next configured plugin
```

Comment:
I don't see anything wrong logically here, but there is a bit of a smell with having an individual operator be in charge of making this determination. I wonder if this would be something better managed at the pipeline level?

Reply:
The pipeline cannot treat each operator equally. Operators that don't have outputs (output plugins) can't be treated the same as inputs or transformers, nor can all transformers be treated the same (router). So, even though I had the same initial reaction, doing it "at the pipeline level" would still require a plugin-specific "SetNextPlugin()" method added to the interface, which would dirty up the `Plugin` interface unncessarily when we have all the information we need in `SetOutputs`

Reply:
This doesn't need to hold up the PR. I don't agree that this would necessarily dirty up the plugin interface or think that it would be very difficult to manage the few types of special cases, but I'm fine with tabling it for now.